.github/workflows/version-bump-and-fast-forward-check.yml

```diff
@@ -0,0 +1,101 @@
+name: Version-bump-and-fast-forward-check
+on: [pull_request]
+
+jobs:
+  check-version-bump:
+    name: check-version-bump
+    runs-on: ubuntu-latest
+    steps:
+      - name: Switch to the branch of the pull request
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          path: ion-java-new
+
+      - name: Checkout the current repository master branch
+        uses: actions/checkout@v3
+        with:
+          repository: amzn/ion-java
+          ref: master
+          path: ion-java-current
+
+      - name: Check whether the pull request contains version number bump
+        run: |
+          cd /home/runner/work/ion-java/ion-java/ion-java-new
+          versionNew=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+          commitID=$(git rev-parse HEAD)
+          cd /home/runner/work/ion-java/ion-java/ion-java-current
+          versionCurrent=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+          if [[ $versionNew != $versionCurrent ]]; then echo "version bump check pass"; else exit 1; fi 
+
+  comment-when-version-number-check-failed:
+    needs: check-version-bump
+    if: ${{ failure() }}
+    runs-on: ubuntu-latest
+    name: comment-when-version-number-check-failed
+    steps:
+      - name: Create comment
+        uses: peter-evans/create-or-update-comment@v2
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            This PR doesn't contain version number bump and will not trigger the ion-java GitHub prerelease process.
```

Comment:
> This is the normal case, correct? For example, this will happen when someone creates a PR for a bug fix. In this case, I'd expect that no comment would be posted and the workflow would succeed immediately without running the fast-forward check.

Yes. That sounds more reasonable to me, I'll update this in the following commit.